okhttp/src/main/java/okhttp3/internal/UtilKt.kt

```diff
@@ -36,3 +40,37 @@ fun BufferedSource.readMedium(): Int {
       or (readByte() and 0xff shl 8)
       or (readByte() and 0xff))
 }
+
+fun Socket.connectionName(): String {
+  val address = remoteSocketAddress
+  return if (address is InetSocketAddress) address.hostName else address.toString()
+}
+
+/** Run [block] until it either throws an [IOException] or completes. */
+inline fun ignoreIoExceptions(block: () -> Unit) {
+  try {
+    block()
+  } catch (_: IOException) {
+  }
+}
+
+/** Execute [block], setting the executing thread's name to [name] for the duration. */
+fun Executor.execute(name: String, block: () -> Unit) {
+  execute(object : NamedRunnable("%s", name) {
+    override fun execute() {
+      block()
+    }
+  })
+}
+
+/** Executes [block] unless this executor has been shutdown, in which case this does nothing. */
+fun Executor.tryExecute(name: String, block: () -> Unit) {
```

Comment:
`inline` + `crossinline`, re-use function above in implementation

Reply:
Done!